### PR TITLE
Add getblocksubsidy RPC command to return block reward taking into account mining slow start

### PR DESCRIPTION
**Pull request by bitcartel**
This PR adds a new RPC command to return the block reward as defined by function `GetBlockSubsidy`.

Usage:
`zcash-cli getblocksubsidy blockheight
`

The basis for this PR is that some users have been unaware of the mining slow start and they subsequently sought help to clarify if they were mining blocks correctly or if there was a bug in the reward schedule.

### CODE REVIEW

File: src/rpcmining.cpp [hunk 1]
```diff
@@ -766,3 +766,27 @@ Value estimatepriority(const Array& params, bool fHelp)
 
     return mempool.estimatePriority(nBlocks);
 }
+
+Value getblocksubsidy(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "getblocksubsidy index\n"
```
Comment (daira) [inline src/rpcmining.cpp]: Call it "height".

File: src/rpcmining.cpp [hunk 2]
```diff
@@ -766,3 +766,27 @@ Value estimatepriority(const Array& params, bool fHelp)
...
+            "\nArguments:\n"
+            "1. index         (numeric, required) The block index\n"
+            "\nResult:\n"
+            "amount           (numeric) The block reward amount in ZEC.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getblocksubsidy", "1000")
+            + HelpExampleRpc("getblockubsidy", "1000")
+        );
+
+    LOCK(cs_main);
```
Comment (daira) [inline src/rpcmining.cpp]: Narrow the lock scope to just before the call to ``GetBlockSubsidy``.

**Comment by daira** [comment]
ACK modulo comments.

File: src/rpcmining.cpp (hunk 2 above)
Comment (bitcartel) [inline src/rpcmining.cpp]: I moved the lock to just before chainActive.tip() to match existing (i.e. upstream) usage.

File: src/rpcmining.cpp (hunk 1 above)
Comment (bitcartel) [inline src/rpcmining.cpp]: Done.

**Comment by ebfull** [comment]
Cool! But should the command show the subsidy minus the founders reward?

**Comment by bitcartel** [comment]
Probably okay for now.  For a future release, we could display how the block subsidy is divided up via an optional flag such as verbose=true which returns {'miningreward':x, 'foundersreward':y}

**Comment by daira** [comment]
I'm generally -1 on optional flags, unless they actually change behaviour (and I'm also sometimes against them then), rather than giving more information that is easy to compute.

**Comment by zookozcash** [comment]
I'm also -1 on optional flags and +0 on showing the founders reward in this RPC output.

**Comment by bitcartel** [comment]
Ok done!

**Comment by ebfull** [comment]
ACK

**Comment by ebfull** [comment]
@zkbot r+

**Comment by ampyx** [comment]
The getblocksubsidy RPC call fails when input is equal to prevHeight+1. This makes the function unusable for mining clients that want to know the subsidy for the next block. 

It would be a matter of changing (rpcmining.cpp:784):

`if (nHeight < 0 || nHeight > chainActive.Height()) `
to
`if (nHeight < 0 || nHeight > (chainActive.Height()+1))`

**Comment by bitcartel** [comment]
@ampyx By design (at the time) to return the subsidy for blocks that have been mined.    I'll open a ticket for this since it's useful to see the subsidy for the next block to be mined.

**Comment by ampyx** [comment]
thanks!

**Pull request merged**